cornellh4i/clean-the-world
# Final data entry flow

## Overview

Closes #29 and #34 : Refactored final data entry flow and implemented modal slide animation on data entry flow. We created a dummy page with dummy text that the user is sent to after confirming new entry (and after the submission modal popup)
The final flow: Home (Login) > Welcome (Add new entry) > DataEntries (new entry -> confirmation -> submissionPopup) > Dummy page

## Changes Made

A lot of the existing code was refactored and new files with new code were added in order to implement the modal slide animation using React Navigation. The router.tsx file creates the root of the modal animation and orders the pages/slides in sequence, the Screen component was created and used in the place of View for each page in order to use React Navigation, and ScreenHeader component was created and used to place the BackButton component at the top. The way we used React Navigation required us to make each frame a separate page in order to make it compatible with the slide animation.

## Test Coverage

1) checkout branch final-data-entry-flow
2) Go through the data entry flow by pressing the Login button on the starting page -> notice the slide animation between every page (except for the new entry page to confirmation page)
3) Try out the Back button on the New Entry page, Confirmation page
4) Temporarily, there is a Back button on the final Dummy page that will bring the user back to the Welcome page with the "Add New Entry" Button

## Next Steps

May or may not need to refactor code to make the animations compatible with the Bottom NavBar
